### ISSUE TITLE
New API

### ISSUE THREAD

**Issue by at-wat** [seqsense/kinesisvideomanager#307]
Switch to Reader/Writer model

```go
type Writer interface {
	Write(*BlockWithBaseTimecode) error
	ReadResponse() (*FragmentEvent, error)
	Close() error
}
func PutMedia(opts ...PutMediaOption) (Writer, error) // immediately return

type Reader interface {
	Read() (*BlockWithBaseTimecode, error)
	ReadTag() (*Tag, error)
	Close() (*Container, error)
}
func GetMedia(opts ...GetMediaOption) (Read, error) // immediately return
```

**Comment by at-wat** [comment]
Partial implementation by https://github.com/seqsense/kinesisvideomanager/pull/305

**Comment by at-wat** [comment]
Now available as v1.0.0-rc.0

**Comment by at-wat** [comment]
Released 1.0.0
